### PR TITLE
Made apiVersion properties a set of variables across a template.

### PR DESCRIPTION
**Pull request by boutquin**
Resubmitting pull request #1080 (which I closed) after adding parameter placeholders to generate unique names; hoping Travis will be happy now.
Changes:
Added variables for apiVersion to visual-studio-dev-vm template; also updated SKUs to most recent versions.
Successfully deployed in US East 2 and US West.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@boutquin__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@boutquin__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

File: visual-studio-dev-vm/azuredeploy.parameters.json [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
-}
+{  
+   "$schema":"http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+   "contentVersion":"1.0.0.1",
+   "parameters":{  
+      "storageAccountNamePrefix":{  
+         "value":"GEN_UNIQUE_11"
+      },
+      "vmIPPublicDnsName":{  
+         "value":"GEN-UNIQUE-13"
```
Comment (sedouard) [inline visual-studio-dev-vm/azuredeploy.parameters.json]: @boutquin this should be GEN_UNIQUE_13

Comment (boutquin) [inline visual-studio-dev-vm/azuredeploy.parameters.json]: If you're referring to the storageAccountNamePrefix parameter, note that I am following @singhkay 's guidance in issue #944.

The parameter isn't used directly, instead a variable gets constructed by appending the resource group id;
"[replace(replace(tolower(concat(parameters('storageAccountNamePrefix'), uniquestring(resourceGroup().id))), '-',''),'.','')]"

That's the variable that gets used for storage names.

Comment (boutquin) [inline visual-studio-dev-vm/azuredeploy.parameters.json]: @sedouard:
Thanks. Your comment led me to the solution: Using dashes instead of underscores.

Comment (sedouard) [inline visual-studio-dev-vm/azuredeploy.parameters.json]: Yeah sorry I meant `GEN-UNIQUE-11`. :-)

**Comment by singhkay** [comment]
@boutquin Looks like the visual-studio-dev-vm was updated today that caused a merge conflict. Can you resolve?

**Comment by boutquin** [comment]
@singhkay 

Happy to! Think I have now incorporated all of @dtzar changes from PR #1090.

**Comment by singhkay** [comment]
@boutquin Looks like there are still more merge conflicts

**Comment by boutquin** [comment]
Merge conflicts resolved, as far as I can tell. Successfully deployed template in US East 2.

**Comment by singhkay** [comment]
@boutquin Take a look at eliminating the storage account from the template. See new guidance at #944. That should also get rid of the error message above

**Comment by boutquin** [comment]
@singhkay 
Thanks; did not notice updated guidance --followed.

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**